index.js

```diff
@@ -755,11 +765,11 @@ Router.prototype.all = function (path, handler, store) {
 module.exports = Router
 
 function escapeRegExp (string) {
-  return string.replace(/[.*+?^${}()|[\]\\]/g, '\\$&')
+  return string.replace(ESCAPE_REGEXP, '\\$&')
 }
 
 function removeDuplicateSlashes (path) {
-  return path.replace(/\/\/+/g, '/')
+  return path.includes('//') ? path.replace(REMOVE_DUPLICATE_SLASHES_REGEXP, '/') : path
```

Comment:
Good find, last time we checked, indexOf was even faster though, can you verify in a vanilla benchmark and use indexOf if it's still the case

```suggestion
  return path.indexOf('//') !== -1 ? path.replace(REMOVE_DUPLICATE_SLASHES_REGEXP, '/') : path
```

Reply:
https://www.programiz.com/online-compiler/1yAOm28DidPxU

Sorry I'm on my phone right now so it's LLM generated but on my phone indexOf still seems faster

Reply:
It seems the `indexOf` is faster in provided benchmark.

Made few runs, but results are not consistent.

```
lookup root "/" route.................................... x 48,849,611 ops/sec ±0.59% (590 runs sampled)
lookup short static route................................ x 31,869,935 ops/sec ±0.31% (593 runs sampled)
lookup long static route................................. x 9,872,279 ops/sec ±0.21% (595 runs sampled)
lookup long static route (common prefix)................. x 7,375,776 ops/sec ±0.17% (596 runs sampled)
lookup short parametric route............................ x 22,025,078 ops/sec ±0.26% (594 runs sampled)
lookup long parametric route............................. x 13,295,798 ops/sec ±0.24% (595 runs sampled)
lookup short parametric route (encoded unoptimized)...... x 13,506,932 ops/sec ±0.22% (594 runs sampled)
lookup short parametric route (encoded optimized)........ x 9,442,262 ops/sec ±0.22% (595 runs sampled)
lookup parametric route with two short params............ x 11,268,643 ops/sec ±0.25% (594 runs sampled)
lookup multi-parametric route with two short params...... x 9,925,770 ops/sec ±0.25% (595 runs sampled)
lookup multi-parametric route with two short regex params x 10,644,109 ops/sec ±0.31% (593 runs sampled)
lookup long static + parametric route.................... x 5,914,833 ops/sec ±0.23% (595 runs sampled)
lookup short wildcard route.............................. x 25,236,359 ops/sec ±0.32% (593 runs sampled)
lookup long wildcard route............................... x 13,000,022 ops/sec ±0.22% (594 runs sampled)
lookup root route on constrained router.................. x 41,795,662 ops/sec ±0.47% (590 runs sampled)
lookup short static unconstraint route................... x 29,526,635 ops/sec ±0.30% (592 runs sampled)
lookup short static versioned route...................... x 23,048,299 ops/sec ±0.34% (592 runs sampled)
lookup short static constrained (version & host) route... x 18,812,944 ops/sec ±0.23% (595 runs sampled)
```

```
lookup root "/" route.................................... x 49,401,378 ops/sec ±0.54% (591 runs sampled)
lookup short static route................................ x 31,764,770 ops/sec ±0.36% (593 runs sampled)
lookup long static route................................. x 9,887,793 ops/sec ±0.19% (595 runs sampled)
lookup long static route (common prefix)................. x 7,387,471 ops/sec ±0.13% (596 runs sampled)
lookup short parametric route............................ x 22,034,581 ops/sec ±0.25% (594 runs sampled)
lookup long parametric route............................. x 13,271,997 ops/sec ±0.24% (594 runs sampled)
lookup short parametric route (encoded unoptimized)...... x 13,500,656 ops/sec ±0.24% (595 runs sampled)
lookup short parametric route (encoded optimized)........ x 9,363,792 ops/sec ±0.20% (594 runs sampled)
lookup parametric route with two short params............ x 11,324,229 ops/sec ±0.22% (594 runs sampled)
lookup multi-parametric route with two short params...... x 9,625,379 ops/sec ±0.25% (591 runs sampled)
lookup multi-parametric route with two short regex params x 10,780,306 ops/sec ±0.25% (594 runs sampled)
lookup long static + parametric route.................... x 6,143,567 ops/sec ±0.15% (595 runs sampled)
lookup short wildcard route.............................. x 25,526,762 ops/sec ±0.29% (593 runs sampled)
lookup long wildcard route............................... x 13,150,146 ops/sec ±0.19% (595 runs sampled)
lookup root route on constrained router.................. x 42,107,307 ops/sec ±0.46% (591 runs sampled)
lookup short static unconstraint route................... x 29,632,898 ops/sec ±0.33% (592 runs sampled)
lookup short static versioned route...................... x 23,092,984 ops/sec ±0.31% (592 runs sampled)
lookup short static constrained (version & host) route... x 18,794,447 ops/sec ±0.18% (594 runs sampled)
```

Reply:
Yeah honestly, the difference will shrink as time goes on so it's not crucially important

Just wanted to mention it :)